xblockmufi/private/edit.js

```diff
@@ -34,4 +36,33 @@ function XblockMufiEdit(runtime, element) {
         });
         return false;
     });
+
+    // TEXT EDITOR JS
+    $('li.specialButton').click(function() {
+        $(this).toggleClass('active');
+        $('.sr .toggle-open', this).toggle();
+        $('.specialCharTable', $element).toggle();
+    });
+
+    $('.txtEditor li.styleButton').mousedown(function(){
+        var command = $(this).data('command');
+        var argument = $(this).data('argument');
+        document.execCommand(command, false, argument);
+        $('.contents', $element).focus();
+    });
+
+    $('.specialCharTable .charWrapper .char').hover(function() {
+        var selChar = $(this).html();
+        var selCharTitle = $(this).attr('title');
+        if (!selCharTitle) {
+            selCharTitle = "";
+        }
+        $('.charZoom').html(selChar);
+        $('.charZoomTitle').text(selCharTitle);
+    });
+
+    $('.specialCharTable .charWrapper .char').mousedown(function() {
+        var textToInsert = $(this).html();
+        document.execCommand('insertHTML', false, textToInsert);
```

Comment:
the function document.execCommand has various things it can do, it can affect the style of the selected text or in this case "insertHTML" at the cursor.